### PR TITLE
session: validate and change tidb_read_staleness input

### PR DESCRIPTION
**Pull request by Yisaer**
Problem Summary:

### What is changed and how it works?

Change and validate input of `tidb_read_staleness`, also add log when there is no target for `closest_replicas`

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -295,6 +296,9 @@ func (a *ExecStmt) RebuildPlan(ctx context.Context) (int64, error) {
 	a.SnapshotTS = ret.LastSnapshotTS
 	a.IsStaleness = ret.IsStaleness
 	a.ReplicaReadScope = ret.ReadReplicaScope
+	if a.Ctx.GetSessionVars().GetReplicaRead().IsClosestRead() && a.ReplicaReadScope == kv.GlobalReplicaScope {
+		logutil.BgLogger().Warn(fmt.Sprintf("tidb can't read closest replicas due to it haven't %s label", placement.DCLabelKey))
```
Comment (JmPotato) [inline executor/adapter.go:300]: Will this cause a massive amount of logs to be printed if the QPS is high?

File: executor/stale_txn_test.go [hunk 2]
```diff
@@ -1122,7 +1122,7 @@ func (s *testStaleTxnSerialSuite) TestStaleSessionQuery(c *C) {
 	tk.MustExec("insert into t10 (id) values (1)")
 	time.Sleep(2 * time.Second)
 	now := time.Now()
-	tk.MustExec(`set @@tidb_read_staleness="-1s"`)
+	tk.MustExec(`set @@tidb_read_staleness="-1"`)
```
Comment (JmPotato) [inline executor/stale_txn_test.go:1125]: So the minimal unit of this session variable is second?

Comment (Yisaer) [reply executor/stale_txn_test.go:1125]: right.

File: executor/adapter.go (hunk 1 above)
Comment (Yisaer) [reply executor/adapter.go:300]: It can be, but `replica-read` only affect session query, the log won't spread for the whole server.

**Review by JmPotato — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**